modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,360 @@
+	 * @param int $floor The lowest possible value.
+	 * @param int $ceil The highest possible value.
+	 */
+	function sanitize_range( $n, $floor, $ceil ) {
+		$n = (int) $n;
+		if ( $n < $floor )
+			$n = $floor;
+		else if ( $n > $ceil )
+			$n = $ceil;
+		return $n;
+	}
+
+	/*
+	 * Sanitize an instance of this widget.
+	 *
+	 * Date ranges match the documentation for mktime in the php manual.
+	 * @see http://php.net/manual/en/function.mktime.php#refsect1-function.mktime-parameters
+	 *
+	 * @uses Milestone_Widget::sanitize_range().
+	 */
+	function sanitize_instance( $dirty ) {
+		$now = (int) current_time( 'timestamp' );
+
+		$dirty = wp_parse_args( $dirty, array(
+			'title'   => '',
+			'event'   => __( 'The Big Day' ),
+			'message' => __( 'The big day is here.' ),
+			'day'     => date( 'd', $now ),
+			'month'   => date( 'm', $now ),
+			'year'    => date( 'Y', $now ),
+			'hour'    => 0,
+			'min'     => 0,
+		) );
+
+		$allowed_tags = array(
+			'a'      => array( 'title' => array(), 'href' => array() ),
+			'em'     => array( 'title' => array() ),
+			'strong' => array( 'title' => array() ),
+		);
+
+		$clean = array(
+			'title'   => trim( strip_tags( stripslashes( $dirty['title'] ) ) ),
+			'event'   => trim( strip_tags( stripslashes( $dirty['event'] ) ) ),
+			'message' => wp_kses( $dirty['message'], $allowed_tags ),
+			'year'    => $this->sanitize_range( $dirty['year'],  1901, 2037 ),
+			'month'   => $this->sanitize_range( $dirty['month'], 1, 12 ),
+			'hour'    => $this->sanitize_range( $dirty['hour'],  0, 23 ),
+			'min'     => zeroise( $this->sanitize_range( $dirty['min'], 0, 59 ), 2 ),
+		);
+
+		$clean['day'] = $this->sanitize_range( $dirty['day'], 1, date( 't', mktime( 0, 0, 0, $clean['month'], 1, $clean['year'] ) ) );
+
+		return $clean;
+	}
+
+    /**
+     * Form
+     */
+    function form( $instance ) {
+		$instance = $this->sanitize_instance( $instance );
+        ?>
+
+	<div class="milestone-widget">
+        <p>
+        	<label for="<?php echo $this->get_field_id( 'title' ); ?>"><?php _e( 'Title' ); ?></label>
+        	<input class="widefat" id="<?php echo $this->get_field_id( 'title' ); ?>" name="<?php echo $this->get_field_name( 'title' ); ?>" type="text" value="<?php echo esc_attr( $instance['title'] ); ?>" />
+        </p>
+
+        <p>
+        	<label for="<?php echo $this->get_field_id( 'event' ); ?>"><?php _e( 'Event' ); ?></label>
+        	<input class="widefat" id="<?php echo $this->get_field_id( 'event' ); ?>" name="<?php echo $this->get_field_name( 'event' ); ?>" type="text" value="<?php echo esc_attr( $instance['event'] ); ?>" />
+        </p>
+
+		<fieldset>
+			<legend><?php _e( 'Date and Time' ); ?></legend>
+
+			<label for="<?php echo $this->get_field_id( 'month' ); ?>" class="assistive-text"><?php _e( 'Month' ); ?></label>
+			<select id="<?php echo $this->get_field_id( 'month' ); ?>" class="month" name="<?php echo $this->get_field_name( 'month' ); ?>"><?php
```

Comment:
When opening a php block with `<?php` probably should put that on a new line, not an awkward trailing bit on the prior line end that's apt to be missed.